### PR TITLE
Added IsDue voter and fixed timeout for long-running scheduled command

### PR DESCRIPTION
**Pull request by TheGrimmChester**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| License       | MIT

Added IsDue voter to allow better check when using cloud service that run cron only every X minutes even if cron is set to run every minutes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,6 +59,11 @@ install-sylius:
 	${YARN} build
 	${CONSOLE} cache:clear
 
+configure-sylius:
+	cd ${TEST_DIRECTORY} && echo '    Synolia\SyliusSchedulerCommandPlugin\Checker\SoftLimitThresholdIsDueChecker:' >> config/services.yaml
+	cd ${TEST_DIRECTORY} && echo '        tags:' >> config/services.yaml
+	cd ${TEST_DIRECTORY} && echo '            - { name: !php/const Synolia\SyliusSchedulerCommandPlugin\Checker\IsDueCheckerInterface::TAG_ID }' >> config/services.yaml
```
Comment (oallain) [inline Makefile:65]: A file should be created in tests/data/config/ with this data and he will be automaticly copied.
Example : https://github.com/synolia/SyliusSchedulerCommandPlugin/blob/master/tests/data/config/packages/test/sylius_fixtures.yaml

File: src/Grid/FieldType/ScheduledCommandExecutionTimeType.php [hunk 2]
```diff
@@ -19,28 +19,29 @@ class ScheduledCommandExecutionTimeType implements FieldTypeInterface
...
 
-        $time = $scheduleCommand->getCommandEndTime()->getTimestamp() - $scheduleCommand->getExecutedAt()->getTimestamp();
+        $baseDateTime = $scheduleCommand->getCommandEndTime() ?? new \DateTime();
+        $time = $baseDateTime->getTimestamp() - $scheduleCommand->getExecutedAt()->getTimestamp();
 
         if ($time > self::HOUR_IN_SECONDES) {
             $hours = (int) ($time / self::HOUR_IN_SECONDES) . 'h ';
-            $minutes = (($time % self::HOUR_IN_SECONDES) / self::MINUTE_IN_SECONDES) . 'm ';
-            $secondes = (($time % self::HOUR_IN_SECONDES) % self::MINUTE_IN_SECONDES) . 's';
+            $minutes = (int) (($time % self::HOUR_IN_SECONDES) / self::MINUTE_IN_SECONDES) . 'm ';
```
Comment (oallain) [inline src/Grid/FieldType/ScheduledCommandExecutionTimeType.php:31]: That fix this mistake ?
image

**Review by oallain — CHANGES_REQUESTED** [review]

File: src/Grid/FieldType/ScheduledCommandExecutionTimeType.php (hunk 2 above)
Comment (TheGrimmChester) [reply src/Grid/FieldType/ScheduledCommandExecutionTimeType.php:31]: yes

File: Makefile (hunk 1 above)
Comment (oallain) [reply Makefile:65]: My bad, in defautl Kernel files under config/services/* hare not loaded

**Review by oallain — APPROVED** [review]

**Comment by oallain** [comment]
Thanks @TheGrimmChester :rocket:

**Pull request merged**